src/mumble/Overlay.cpp

```diff
@@ -212,7 +212,13 @@ Overlay::~Overlay() {
 }
 
 void Overlay::setActive(bool act) {
+	if (!act && !m_initialized.load()) {
+		// Disabling when the Overlay hasn't been initialized yet, doesn't make much sense
+		return;
+	}
```

Comment:
**suggestion:** Personally I think I would make the init `if (act && !m_initialized)` instead. This block introduces another condition, negated and combined, that was difficult for me to wrap my head around. I think "init if necessary" would be clearer than this "special special" condition.

Reply:
I'm also preventing a call to `setActiveInternal` as imo we shouldn't call internal functions, if the internals have not been initialized yet.

Plus I'm a big fan of putting exit conditions first in a function so that you don't have to wrap your actual code into if/else-Blocks. In this case it'd only be a single block but this can easily lead to obfuscation of the actual code (imo)

Reply:
I generally prefer early returns as well. But with the overall state and stuff it was difficult for me to wrap my head around here. It was harder instead of easier to interpret the function.

Reply:
Well we could also rewrite the condition to
```c
if (!(act && m_initialized.load()))
```
if you think that helps...

Reply:
Your call.

Reply:
I didn't like all alternatives I could come up with, so I left it unchanged. I think in combination with the comment, it should be somewhat clear what this is about :thinking: